l10n_id_taxform_period/models/tax_period.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Prime Force Indonesia
+# Copyright 2016 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+
+class TaxYear(models.Model):
+    _name = "l10n_id.tax_year"
+    _description = "Tax Year"
+    _order = "date_start asc, id"
+
+    name = fields.Char(
+        string="Tax Year",
+        required=True,
+    )
+    code = fields.Char(
+        string="Code",
+        required=True,
+    )
+    date_start = fields.Date(
+        string="Date Start",
+        required=True,
+    )
+    date_end = fields.Date(
+        string="Date End",
+        required=True,
+    )
+    period_ids = fields.One2many(
+        string="Periods",
+        comodel_name="l10n_id.tax_period",
+        inverse_name="year_id",
+    )
+
+    @api.constrains("date_start", "date_end")
+    def _check_range(self):
+        if self.date_end <= self.date_start:
+            strWarning = _("The start date must precede it's end date")
+            raise models.ValidationError(strWarning)
```

Comment:
Hi @moylop260 . Many thanks for your reviews.

I use wrapper form models because i don't have to import additional module. Does importing from exceptions give technically better result than from models?